src/GraphQL.DataLoader/DataLoaderContextAccessor.cs

```diff
@@ -6,9 +6,14 @@ public class DataLoaderContextAccessor : IDataLoaderContextAccessor
     private static readonly AsyncLocal<DataLoaderContext?> _current = new();
 
     /// <inheritdoc/>
-    public DataLoaderContext? Context
+    [AllowNull]
+    public DataLoaderContext Context
     {
-        get => _current.Value;
+        get => _current.Value ?? ThrowMissingContext();
         set => _current.Value = value;
     }
+
+    [DoesNotReturn]
+    private DataLoaderContext ThrowMissingContext()
+        => throw new InvalidOperationException("DataLoaderContext is null. Ensure that DataLoaderDocumentListener is registered in the IoC container");
```

Comment:
A few comments here:
- By default, neither GraphQL.NET nor GraphQL.NET Server pull all `IDocumentExecutionListener` instances from DI, so registering the listener in the IoC container is not enough to fix the problem.  While it is possible with custom code to do so, the current suggested mechanism is to use `.AddDataLoaders()` to install the data loader document listener.  This will register `DataLoaderDocumentListener` in the DI container and install a `IConfigureExecution` instance which is set up to pull the listener from DI and add it to `ExecutionOptions.Listeners`.
- By convention, we end all exception messages with a period.

```suggestion
        => throw new InvalidOperationException("DataLoaderContext is null. Ensure that you have called IGraphQLBuilder.AddDataLoader() or otherwise have added an instance of DataLoaderDocumentListener to ExecutionOptions.Listeners.");
```